lib/msal-core/src/utils/WindowUtils.ts

```diff
@@ -107,6 +107,23 @@ export class WindowUtils {
         });
     }
 
+    /**
+     * @hidden
+     * Loads the iframe synchronously when the navigateTimeFrame is set to `0`
+     * @param urlNavigate
+     * @param frameName
+     * @param logger
+     */
+    static syncLoadFrame(urlNavigate: string, frameName: string, logger: Logger): HTMLIFrameElement{
+        const frameHandle = WindowUtils.addHiddenIFrame(frameName, logger);
+        if (frameHandle.src === "" || frameHandle.src === "about:blank") {
+            frameHandle.src = urlNavigate;
+            logger.infoPii("Frame Name : " + frameName + " Navigated to: " + urlNavigate);
```

Comment:
And you could have the async version call the sync version inside the `setTimeout`, to make it so the logic of setting the `src` isn't duplicated.